packages/dds/tree/src/jsonDomainSchema.ts

```diff
@@ -0,0 +1,110 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	SchemaFactory,
+	type AllowedTypes,
+	type FixRecursiveArraySchema,
+	type TreeNodeFromImplicitAllowedTypes,
+	type ValidateRecursiveSchema,
+} from "./simple-tree/index.js";
+
+const sf = new SchemaFactory("com.fluidframework.json");
+
+/**
+ * {@link AllowedTypes} for primitives types allowed in JSON.
+ * @alpha
+ */
+export const JsonPrimitive = [
+	sf.null,
+	sf.boolean,
+	sf.number,
+	sf.string,
+] as const satisfies AllowedTypes;
+
+/**
+ * @alpha
+ */
+export type JsonPrimitive = TreeNodeFromImplicitAllowedTypes<typeof JsonPrimitive>;
+
+/**
+ * {@link AllowedTypes} for any content allowed in the JSON domain.
+ * @example
+ * ```typescript
+ * const tree = TreeAlpha.importConcise(JsonUnion, { example: { nested: true }, value: 5 });
+ * ```
+ * @alpha
+ */
+export const JsonUnion = [() => JsonObject, () => JsonArray, ...JsonPrimitive] as const;
```

Comment:
My vote would be to call this `Json` or `JsonTree`. The term "Union", while correct, is more mechanical than it is semantic.

Reply:
If we want to be more careful / explicit with the naming, we could possibly do something like `JsonDomainTree`. I agree that `JsonUnion` doesn't really convey the semantics we want here.

Reply:
I don't think JsonDomain really does it either. Its perfectly reasonable to think of VerboseTree and ConciseTree as projections of trees into the JSON domain. This format on the other hand isn't even a JSON style object since it uses maps for objects.

Maybe `JsonAsTree` for JsonUnion, and JsonObjectAsTree for JsonObject?

Another approach would be to go with a namespace. Since everything in it needs the same release tag, I think we can make that work. In that case maybe JsonDomain, with a big doc comment on the namespace, would work, then the members could have short names like Object, Union/Tree, Array, etc.?

Reply:
Or, well, maybe not, since we're generating our own rollups now. I'm not sure.

Reply:
But I do like namespaces for things like this. `JsonAsTree` also seems reasonable to me.